src/Rtc/Create.tsx

```diff
@@ -0,0 +1,138 @@
+import React, {useState, useEffect, useContext, useRef, FC} from 'react';
+import RtcEngine, {
+  VideoEncoderConfiguration,
+  AreaCode,
+} from 'react-native-agora';
+import {Platform} from 'react-native';
+import requestCameraAndAudioPermission from '../Utils/permission';
+import {DispatchType} from '../Contexts/RtcContext';
+import PropsContext, {ToggleState} from '../Contexts/PropsContext';
+import quality from '../Utils/quality';
+
+const Create = ({
+  dispatch,
+  children,
+}: {
+  dispatch: DispatchType;
+  children: (engine: React.MutableRefObject<RtcEngine>) => JSX.Element;
+}) => {
+  const [ready, setReady] = useState(false);
+  const {callbacks, rtcProps} = useContext(PropsContext);
+  let engine = useRef<RtcEngine>({} as RtcEngine);
+
+  useEffect(() => {
+    async function init() {
+      if (Platform.OS === 'android') {
+        //Request required permissions from Android
+        await requestCameraAndAudioPermission();
+      }
+      try {
+        if (Platform.OS === 'android' || Platform.OS === 'ios') {
+          engine.current = await RtcEngine.createWithAreaCode(
+            rtcProps.appId,
+            // eslint-disable-next-line no-bitwise
+            AreaCode.GLOB ^ AreaCode.CN,
+          );
+        } else {
+          engine.current = await RtcEngine.create(rtcProps.appId);
+        }
+        if (rtcProps.profile) {
+          if (Platform.OS === 'web') {
```

Comment:
Should we move this to the bridge instead of the UIKit?

Reply:
We won't be moving this as per discussions yday